### PR TITLE
Add timer for data fetching across the project

### PR DESCRIPTION
**Pull request by MayoG**

### CODE REVIEW

File: components/routine-logs-view.js [hunk 1]
```diff
@@ -18,13 +21,17 @@ const columns = [
...
+export default function RoutineLogsView({ logsPerPage }) {
   const selectedRoutine = useRecoilValue(selectedRoutineState);
   const [open, setOpen] = useState(false);
   const [time, setTime] = useState("");
   const [level, setLevel] = useState("");
   const [message, setMessage] = useState("");
-  const [logs, setLogs] = useState([]);
+
+  const { data } = useSWR(selectedRoutine ? `routine_logs/${selectedRoutine}` : null, apiFetch, {
+    refreshInterval: 5000,
```
Comment (tomerdayan168) [inline components/routine-logs-view.js]: Move to const

File: utils/http-calls.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { SERVER_URL } from '../config';
+
+export async function apiFetch(endpoint) {
+    let response = await fetch(`${SERVER_URL}/api/${endpoint}`);
+
+    if (response.status >= 400 && response.status < 600) {
```
Comment (tomerdayan168) [inline utils/http-calls.js]: There are libraries that contains those error codes and their name, use that instead of writing 400 and 600

**Review by tomerdayan168 — CHANGES_REQUESTED** [review]
:woman_beard:

File: components/routine-logs-view.js (hunk 1 above)
Comment (San-Moshe) [reply components/routine-logs-view.js]: I'd suggest allowing the user to configure it via UI. 
What do you think?

File: utils/http-calls.js (hunk 2 above)
Comment (MayoG) [reply utils/http-calls.js]: Used response.ok

File: components/routine-logs-view.js (hunk 1 above)
Comment (MayoG) [reply components/routine-logs-view.js]: IMO I don't think we should implement it if we don't need it yet :)

Comment (San-Moshe) [reply components/routine-logs-view.js]: Yes, it might be better to implement it as a separate feature.

File: utils/http-calls.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { SERVER_URL } from '../config';
+
+export async function apiFetch(endpoint) {
+    let response = await fetch(`${SERVER_URL}/api/${endpoint}`);
+
+    if (response.ok) {
+        return response.json()
+        
+    } else 
+        throw `An error when fetching from /api/${endpoint} - ${response.text}`
```
Comment (Elon-Abulafia) [inline utils/http-calls.js:10]: Why aren't you using `.then()` and `.catch()`?

Comment (San-Moshe) [reply utils/http-calls.js:10]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#rewriting_a_promise_chain_with_an_async_function

**Review by tomerdayan168 — APPROVED** [review]

**Pull request merged**